File: 1547.py
n = int(input())
for i in range(n):
    e = [int(x) for x in str(input()).split()]
    qt = e[0]
    s = e[1]
    e = [int(x) for x in str(input()).split()]
    d = ind = 999
    for j in range(qt):
        if d > abs(s - e[j]):
            d = abs(s - e[j])
            ind = j
    print(ind + 1)

File: 2-ad-hoc/1547.cpp
#include <iostream>

using namespace std;

int main(void) {
    int n, i, j, qt, s, d, m, ind;

    cin >> n;
    for (i = 0; i < n; ++i) {
        cin >> qt >> s;
        d = ind = 999;
        for (j = 0; j < qt; ++j) {
            cin >> m;
            if (d > abs(s - m)) {
                d = abs(s - m);
                ind = j;
            }
        }
        cout << ind + 1 << endl;
    }
    return 0;
}
